src/components/CurrentImage/index.jsx

```diff
@@ -0,0 +1,128 @@
+  const [isSolved, setIsSolved] = useState(false);
+  const tilesGenerator = () => {
+    const newTilesArray = [];
+    // for loop to generate tiles where tiles are puzzle pieces
+    const numOfTiles = 6;
+    for (let tileId = 0; tileId < numOfTiles; tileId++) {
+      let tileTopStyling = tileId * 100;
+      // value of selected is used for styling purposes where the selected tile should be highlighted
+      // value of top is used for fitting the image inside the tile
+      let newTile = { id: tileId, selected: false, top: tileTopStyling };
+      newTilesArray.push(newTile);
+    }
+    // for loop that generates a random and extracts the element with the index
+    // equal to the random number, then pushes that element to the end of newTilesArray.
+    for (let i = 0; i < newTilesArray.length; i++) {
+      let random = Math.floor(Math.random() * (newTilesArray.length - 1));
+      let randomItemArray = newTilesArray.splice(random, 1);
+      newTilesArray.push(randomItemArray[0]);
+    }
+    // Set the randomized array as generated tiles
+    setGeneratedTiles(newTilesArray);
+  };
+
+  useEffect(() => {
+    tilesGenerator();
+  }, []);
+
+  useEffect(() => {
+    checkSolution();
+  }, [generatedTiles]);
+
+  const handleSelect = (currId) => {
+    // if selectedTile's value is not equal to null (meaning there a tile seleted already)
+    // then when another tile is clicked swap both tiles' places
+    // else set the selectedTile to the id of the clicked tile
+    if (selectedTile !== null) {
+      const prevId = selectedTile;
+      swapTiles(prevId, currId);
+    } else {
+      setSelectedTile(currId);
+      const tileIndex = generatedTiles.findIndex((tile) => tile.id === currId);
+      // set the select value of the tile to true
+      generatedTiles[tileIndex] = {
+        ...generatedTiles[tileIndex],
+        selected: true,
+      };
+    }
+  };
+
+  const swapTiles = (id0, id1) => {
+    let newTiles = [...generatedTiles];
+    let index0 = generatedTiles.findIndex((tile) => tile.id === id0);
+    let index1 = generatedTiles.findIndex((tile) => tile.id === id1);
+    let tile0 = { ...generatedTiles[index0] };
+    let tile1 = { ...generatedTiles[index1] };
+    newTiles[index0] = tile1;
+    newTiles[index1] = tile0;
+    newTiles.forEach((tile) => (tile.selected = false));
+    setGeneratedTiles(newTiles);
+    setSelectedTile(null);
+  };
+
+  const checkSolution = () => {
+    // tiles are saved in an array indexed from 0 to n (6) and displayed in that order
+    // first condition checks if tile's id is equal to its index in the array and returns false/true
+    // since initial acc value is true, second condition checks if prev value is true and keeps its value
+    const isCorrect = generatedTiles.reduce((acc, curr, index) => {
+      if (curr.id !== index) {
+        return false;
+      } else if (acc === true) {
+        return acc;
+      }
```

Comment:
still seems like 85 to 89 can be rewritten 

its generally dangerous to have code that has a case that has no return value. here, you have an if and an else-if; two conditions. my question is, is there a third case where neither of those is true, and this code is just returning nothing? (the conditionals should be structured so i dont have to ask this question after i read it)

Reply:
I fixed the issue and since there are only 2 way that the acc can be false, I included these two conditions at the top and for all the other conditions (else) the result would be true PTAL